### ISSUE TITLE
Extension causes "go to definition" to go to type definition

### ISSUE THREAD

**Issue by kylemh** [mdx-js/mdx-analyzer#461]
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

1.8.9 and styled-components@^5

### Steps to reproduce

- Use Next.js starter
- Install `styled-components@^5`
- Install @types/styled-components`
- Install MDX VS Code extension
- Define some simple component and render it on `app/page.tsx` or `src/app/page.tsx` or `pages/index.tsx` or `src/pages/index.tsx`: a lá:
```ts
const BlueText = styled.p`
  color: blue;
`;
```
- In VS Code, right-click on the "BlueText" part of `<BlueText>Test</BlueText>`
- Click "Go To Definition"
- Be taken to type definitions instead of your inline definition of `BlueText`

Somebody else saw this problem, too: https://github.com/microsoft/vscode/issues/211725

### Expected behavior

I should be taken to source definition, especially with `"typescript.preferGoToSourceDefinition": true`

### Actual behavior

I am taken to type definition

### Runtime

Node v18

### Package manager

pnpm

### OS

macOS

### Build and bundle tools

Next.js

**Comment by remcohaszing** [comment]
I’m unable to reproduce. I tried it with both 1.8.9 and 1.8.10. Does the problem still exist for you? If so, please provide a repo with a minimal reproduction.

**Comment by kylemh** [comment]
The problem existed until I removed the extension! It was happening on a work repo and - once I saw that comment in the TS repo - I attempted the minimum repro that I shared. It's a bit tough for me to share a perfect reproduction because I can't control the Node version of online versions of VS Code. If you did the steps with the correct Node version and told VS Code to use the workspace version of TS (as opposed to the editor version), then I've got no clue! If I re-add, I continue to get the same issue.
